src/main/java/com/synopsys/integration/alert/channel/email/descriptor/EmailDistributionDescriptorActionApi.java

```diff
@@ -72,56 +73,66 @@ public void validateChannelConfig(final FieldAccessor fieldAccessor, final Map<S
         final Set<String> emailAddresses = new HashSet<>();
-        Set<BlackDuckProjectEntity> blackDuckProjectEntities = null;
-
+      
         final Optional<String> filterByProject = fieldModel.getField(BlackDuckDescriptor.KEY_FILTER_BY_PROJECT).getValue();
         final Optional<String> providerName = fieldModel.getField(CommonDistributionUIConfig.KEY_PROVIDER_NAME).getValue();
 
-        if (filterByProject.isPresent() && BooleanUtils.toBoolean(filterByProject.get())) {
-            final Set<String> configuredProjects = fieldModel.getField(BlackDuckDescriptor.KEY_CONFIGURED_PROJECT).getValues().stream().collect(Collectors.toSet());
-            final Optional<String> projectNamePattern = fieldModel.getField(BlackDuckDescriptor.KEY_PROJECT_NAME_PATTERN).getValue();
-            blackDuckProjectEntities = blackDuckProjectRepositoryAccessor.readEntities()
-                                           .stream()
-                                           .map(databaseEntity -> (BlackDuckProjectEntity) databaseEntity)
-                                           .filter(databaseEntity -> emailGroupChannel.doesProjectNameMatchThePattern(databaseEntity.getName(), projectNamePattern.orElse(""))
-                                                                         || emailGroupChannel.doesProjectNameMatchAConfiguredProject(databaseEntity.getName(), configuredProjects))
-                                           .collect(Collectors.toSet());
-        } else if (providerName.isPresent() && BlackDuckProvider.COMPONENT_NAME.equals(providerName.get())) {
-            blackDuckProjectEntities = blackDuckProjectRepositoryAccessor.readEntities()
-                                           .stream()
-                                           .map(databaseEntity -> (BlackDuckProjectEntity) databaseEntity)
-                                           .collect(Collectors.toSet());
+        if (providerName.isPresent() && BlackDuckProvider.COMPONENT_NAME.equals(providerName.get())) {
+            Set<BlackDuckProjectEntity> blackDuckProjectEntities = null;
+            if (filterByProject.isPresent() && BooleanUtils.toBoolean(filterByProject.get())) {
+                final Set<String> configuredProjects = fieldModel.getField(BlackDuckDescriptor.KEY_CONFIGURED_PROJECT).getValues().stream().collect(Collectors.toSet());
+                final Optional<String> projectNamePattern = fieldModel.getField(BlackDuckDescriptor.KEY_PROJECT_NAME_PATTERN).getValue();
+                blackDuckProjectEntities = blackDuckProjectRepositoryAccessor.readEntities()
+                                               .stream()
+                                               .map(databaseEntity -> (BlackDuckProjectEntity) databaseEntity)
+                                               .filter(databaseEntity -> doesProjectNameMatchThePattern(databaseEntity.getName(), projectNamePattern.orElse(""))
```

Comment:
I think we should abstract this filter into its own method. It feels like there is way too much happening in one stream operation. Once abstracted, we could call it like this: `this::isProjectConfigured`